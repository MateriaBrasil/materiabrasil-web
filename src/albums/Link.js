import React from 'react'
import BookmarkIcon from '@material-ui/icons/Bookmark'
import Tooltip from '@material-ui/core/Tooltip'
import Link from 'react-router-dom/Link'
import Colors from '../Colors'

export default props => (
  <Tooltip title={'Adicionar a um album'}>
    <Link
      to={`/materials/${props.id}/albums`}
      style={{
        textDecoration: 'none',
        padding: 12,
        verticalAlign: 'bottom',
        display: 'inline-block',
      }}
    >
      <BookmarkIcon
        style={{
          color: props.white ? Colors.white : Colors.black,
          verticalAlign: 'bottom',
        }}
      />
    </Link>
  </Tooltip>
)
