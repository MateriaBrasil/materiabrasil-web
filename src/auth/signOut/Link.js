import React from 'react';
import Typography from '@material-ui/core/Typography';

import onClick from '../../onClick';

export default ({ destroy, isInstitutionalPage, transparent }) => (
  <a
    className="navbar-link"
    href="#destroy"
    onClick={onClick(destroy)}
    style={{
      marginLeft: 20,
      color: transparent && window.outerWidth > 768 ? '#FFF' : '#000',
    }}
  >
    <Typography
      variant="subtitle1"
      color="inherit"
      style={{ display: 'inline-block', alignSelf: 'center' }}
    >
      Sair
    </Typography>
  </a>
);
