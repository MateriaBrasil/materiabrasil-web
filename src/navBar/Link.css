.navbar-links .navbar-link {
  padding: 10px;
  margin-left: 20px;
  margin-right: 20px;
}

.navbar-links .navbar-link:hover {
  border-bottom: 3px solid black;
}

.menu-appbar .navbar-link {
  padding: 0px;
  width: 100%;
  margin: 0 !important;
}
