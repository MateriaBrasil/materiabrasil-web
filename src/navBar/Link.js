import React from 'react'
import Link from 'react-router-dom/Link'
import Typography from '@material-ui/core/Typography'
import './Link.css'

export default ({ to, text, isInstitutionalPage, targetBlank }) => (
  <Link
    className="navbar-link"
    to={to}
    target={targetBlank ? '_blank' : '_self'}
    style={{
      textDecoration: 'none',
      color: isInstitutionalPage ? 'white' : 'black',
    }}
  >
    <Typography
      variant="subtitle1"
      color="inherit"
      style={{ display: 'inline-block', alignSelf: 'center' }}
    >
      {text}
    </Typography>
  </Link>
)
