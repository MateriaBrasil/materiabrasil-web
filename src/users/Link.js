import React from 'react'
import Link from 'react-router-dom/Link'
import Typography from '@material-ui/core/Typography'

export default ({ currentUser }) => (
  <Link to={`/profile`} style={{ textDecoration: 'none', color: 'black' }}>
    <Typography
      variant="subheading"
      color="inherit"
      style={{ display: 'inline-block', marginRight: 10 }}
    >
      {currentUser.firstName} {currentUser.lastName}
    </Typography>
  </Link>
)
