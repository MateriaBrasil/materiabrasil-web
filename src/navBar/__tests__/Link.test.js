import React from 'react'
import renderer from 'react-test-renderer'

import Link from '../Link'

jest.mock('react-router-dom/Link', () => props => (
  <div {...props}>Link - {props.children}</div>
))

it('renders correctly', () => {
  const tree = renderer.create(<Link to="/foo" text="foo bar" />)
  expect(tree).toMatchSnapshot()
})
