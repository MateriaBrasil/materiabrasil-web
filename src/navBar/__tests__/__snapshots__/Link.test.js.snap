// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar-link"
  style={
    Object {
      "color": "black",
      "textDecoration": "none",
    }
  }
  target="_self"
  to="/foo"
>
  Link - 
  <h6
    className="MuiTypography-root-1 MuiTypography-subtitle1-19 MuiTypography-colorInherit-30"
    style={
      Object {
        "alignSelf": "center",
        "display": "inline-block",
      }
    }
  >
    foo bar
  </h6>
</div>
`;
